src/server/scripts/EasternKingdoms/Gnomeregan/boss_mekgineer_thermaplugg.cpp

```diff
@@ -0,0 +1,464 @@
+                if (GameObject* face = me->FindNearestGameObject(GnomeFaces[i], 130.0f))
+                    face->AI()->Reset();
+        }
+
+        void Reset() override
+        {
+            _events.Reset();
+            Initialize();
+            DespawnBombs();
+        }
+
+        void KilledUnit(Unit* who) override
+        {
+            if (who->GetTypeId() == TYPEID_PLAYER)
+                Talk(SAY_DEAD);
+        }
+
+        void JustDied(Unit* /*killer*/) override
+        {
+            _JustDied();
+            DespawnBombs();
+            for (uint8 i = 0; i < MAX_GNOME_FACES; i++)
+                if (GameObject* face = me->FindNearestGameObject(GnomeFaces[i], 130.0f))
+                    face->AI()->Reset();
+        }
+
+        void JustEnteredCombat(Unit* /*who*/) override
+        {
+            for (uint8 i = 0; i < MAX_GNOME_FACES; i++)
+                if (GameObject* face = me->FindNearestGameObject(GnomeFaces[i], 130.0f))
+                    AddFaceAvailable(GnomeFaces[i]);
+
+            Talk(SAY_AGGRO);
+            _events.ScheduleEvent(EVENT_POUND, 6 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_STEAM_BLAST, 10 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_ACTIVATE_BOMBS, 8 * IN_MILLISECONDS);
+            _events.ScheduleEvent(EVENT_WELDING_BEAM, 14 * IN_MILLISECONDS);
+        }
+
+        void UpdateAI(uint32 diff) override
+        {
+            if (!UpdateVictim())
+                return;
+
+            if (me->HasUnitState(UNIT_STATE_CASTING))
+                return;
+
+            _events.Update(diff);
+
+            while (uint32 eventId = _events.ExecuteEvent())
+            {
+                switch (eventId)
+                {
+                case EVENT_POUND:
+                    if (Unit* target = SelectTarget(SELECT_TARGET_RANDOM, 0, 5.0f))
+                        DoCast(target, SPELL_POUND);
+                    else
+                        DoCastVictim(SPELL_POUND);
+
+                    _events.RescheduleEvent(EVENT_POUND, 10 * IN_MILLISECONDS);
+                    _events.RescheduleEvent(EVENT_STEAM_BLAST, 4 * IN_MILLISECONDS);
+                    break;
+                case EVENT_STEAM_BLAST:
+                {
+                    uint8 rand = urand(1, 4);
+                    if (rand == 1)
+                        Talk(SAY_MACHINES);
+                    DoCastVictim(SPELL_STEAM_BLAST);
+                    if (GetThreat(me->GetVictim()))
+                        ModifyThreatByPercent(me->GetVictim(), -50);
+                }
+                break;
+                case EVENT_ACTIVATE_BOMBS:
+                {
+                    uint32 faceEntry = Trinity::Containers::SelectRandomContainerElement(_availableFacesList);
+                    if (GameObject* face = me->FindNearestGameObject(faceEntry, 125.0f))
+                    {
+                        Talk(SAY_EXPLOSIONS);
+                        face->AI()->DoAction(ACTION_ACTIVATE);
```

Comment:
If you're already calling out to the AI, the `SetGoState` call shouldn't be here

Reply:
can't even find it in spellwork, there's https://wotlk.evowow.com/?spell=11518 but nothing on wowhead either.

Reply:
https://woehead.way-of-elendil.fr/?spell=11518

Reply:
aha, found it in world.spell_dbc